exercises/practice/leap/leap.odin

```diff
@@ -0,0 +1,11 @@
+package leap
+
+Error :: enum {
+	// Implement me!
+	NotImplemented,
+}
+
+is_leap_year :: proc(year: int) -> (bool, Error) {
+	// Implement me!
+	return false, .NotImplemented
+}
```

Comment:
I'm no Odin expert (not even a novice), but while the custom Error setup is likely very idiomatic Odin, I'm not sure it is the easiest for the student to work with. I've read the docs, and maybe using `#panic(message)` is an option here too? I think the stub would then look like:

```suggestion
is_leap_year :: proc(year: int) -> bool {
	#panic("Please implement the `is_leap_year` procedure")
}
```

That's not to say that every exercise must be like this. If an exercise has to explicitly deal with errors, the above approach should of course be used.

Reply:
That's actually a very good idea. Although the `enum` is one way of defining error cases, not all procedures need those (certainly not `is_leap_year`, for which all input would be valid), and Odin [will never have exceptions](https://www.gingerbill.org/article/2018/09/05/exceptions-and-why-odin-will-never-have-them/). The panic is much cleaner. Thanks!